modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc

```diff
@@ -273,3 +273,171 @@ function template_preprocess_ding_ddbasic_opening_hours_all(&$variables) {
     $structured
   );
 }
+
+/**
+ * Categorized library opening hours.
+ */
+function theme_ding_ddbasic_opening_hours_all_tabbed($variables) {
+  return render($variables['tabbed']);
+}
+
+/**
+ * Preprocessing categorized library opening hours.
+ */
+function template_preprocess_ding_ddbasic_opening_hours_all_tabbed(&$variables) {
+  drupal_add_library('system', 'ui.tabs');
+  drupal_add_js('jQuery(document).ready(function(){jQuery("#oh-tabs").tabs();});', 'inline');
+
+  $today = strtotime('today');
+
+  if (!empty($variables['today'])) {
+    $today = $variables['today'];
+  }
+
+  $queue = nodequeue_load_queue_by_name('ding_library_listing');
+  foreach (nodequeue_load_nodes($queue->qid, FALSE, 0, 0) as $node) {
+    $order[$node->title] = $node->nid;
+  }
+
+  // Get and sort all the instances in the given timespan.
+  $instances = opening_hours_instance_load_multiple(
+    array_values($order),
+    date('Y-m-d', $today),
+    date('Y-m-d', $today)
+  );
+
+  foreach ($order as $library_nid) {
```

Comment:
`$order` array is defined in the `nodequeue_load_nodes` foreach call. Is it possible that this($order) array to be empty? If so, I think it is better to define it somewhere before its usage.